hbw/app/javascript/packs/hbw/components/shared/context/store.jsx

```diff
@@ -22,14 +22,67 @@ const withStoreContext = (WrappedComponent) => {
       this.initStore();
     }
 
+    getTaskById = async (data) => {
+      const { env } = this.props;
+      const incomingTask = await env.connection.request({
+        url:  `${env.connection.serverURL}/tasks/${data.task_id}`,
+        data: {
+          cache_key:    data.cache_key,
+          entity_class: env.entity_class,
+        },
+      });
+
+      const tasks = this.state.tasks.filter(task => task.id !== incomingTask.id);
+      tasks.push(incomingTask);
+      this.setState({
+        tasks:    (this.orderTasks(tasks)),
+        fetching: false,
+        ready:    true
```

Comment:
I don't think you should touch `fetching` or `ready` here, these keys should be used for initialization only.